deployment/helm/eclipse-ditto/values.yaml

```diff
@@ -2,135 +2,654 @@
+    ssl: false
 
+## ingress configuration
+ingress:
+  enabled: false
+  annotations: {}
+    # kubernetes.io/tls-acme: "true"
+    # ingress.kubernetes.io/ssl-redirect: "true"
+  hosts:
+    - host: ditto.example.com
+      paths: []
+        # - /api
+        # - /ws
+        # - /devops
+        # - /status
+        # - /health
+        # - /apidoc
+
+  tls: []
+  #  - secretName: ditto-tls
+  #    hosts:
+  #      - ditto.example.com
+
+## ----------------------------------------------------------------------------
+## global configuration shared by all components
+global:
+  ## jwtOnly controls whether only OpenID-Connect authentication is supported
+  ## if false, basicAuth is used
+  ## ref: https://www.eclipse.org/ditto/installation-operating.html#openid-connect
+  jwtOnly: false
+  ## timezone
+  timezone: Europe/Berlin
+  ## imagePullSecrets will be added to every deployment
+  imagePullSecrets: []
+  prometheus:
+    ## enabled controls whether scrape config annotation will be added to pod templates
+    enabled: true
+    ## port where prometheus metrics will be provided
+    port: 9095
+
+
+## ----------------------------------------------------------------------------
+## akka actor configuration
+## ref: https://doc.akka.io/docs/akka/current/typed/index.html
 akka:
+  ## actorSystemName
   actorSystemName: ditto-cluster
   remoting:
+    ## port for remoting
     port: 2551
   mgmthttp:
+    ## port for http managment
     port: 8558
 
-global:
-  timezone: Europe/Berlin
 
+## ----------------------------------------------------------------------------
+## concierge configuration
+## ref: https://www.eclipse.org/ditto/architecture-services-concierge.html
 concierge:
+  ## enabled controls whether Concierge related resources should be created
   enabled: true
-  name: concierge
+  ## replicaCount for concierge service
   replicaCount: 1
-  memory: 512Mi
-  networkPolicy:
-    enabled: true
+  ## updateStrategy for concierge service
+  ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  ## additionalLabels on concierge pods
+  additionalLabels: {}
+  ## additionalAnnotations on concierge pods
+  additionalAnnotations: {}
   image:
+    ## repository for the concierge docker image
     repository: docker.io/eclipse/ditto-concierge
-    imagePullPolicy: IfNotPresent
-    tag: 1.0.0-M1a
+    ## tag for the concierge docker image
+    tag: 1.0.0-M3
+    ## pullPolicy for the concierge docker image
+    pullPolicy: IfNotPresent
+  ## systemProps used to define arbritrary system properties for concierge service
+  ## ref: https://www.eclipse.org/ditto/installation-operating.html#configuration
+  systemProps: []
+  ## extraEnv to add arbritrary environment variable to concierge container
+  extraEnv: []
+  ## resources for the concierge container
+  resources: {}
+    # requests:
```

Comment:
Helm best practice is to disable requests and limits by default but define sane default values.

Reply:
This comment from the template (when creating a chart with `helm create`) explains it:
```
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube.
```

Reply:
ok, so in order to define those "sane default values" - would they have to be placed not in the `*-deployment.yaml` files when not explicitly set in `values.yaml`?

Reply:
Sane default values are still in place (but as comment). Keep in mind, that the values.yaml contains just default values which should work for almost everybody (independently whether it will be installed on a local minikube or into a cluster).

What we could do, is to provide more config files named `values-dev.yaml` or `values-prod.yaml`.

Or we could add some info to the README.

Reply:
I think adding to the Readme should be fine.
Just want to avoid "Ditto uses too much memory" issues for not so experienced k8s/Helm users 😉

Reply:
I added an info to the README